lib/jekyll/filters/url_filters.rb

```diff
@@ -23,9 +23,9 @@ def absolute_url(input)
       def relative_url(input)
         return if input.nil?
         site = @context.registers[:site]
-        return ensure_leading_slash(input.to_s) if site.config["baseurl"].nil?
+        parts = [site.config["baseurl"], input]
```

Comment:
The other method in `master`, [`sanitized_baseurl`](https://github.com/jekyll/jekyll/blob/b0136fccfab51fd7885207fa79bc3dbf3f64c6f4/lib/jekyll/filters/url_filters.rb#L44-L47), always returns a string, which means this code is not valid on `master`. Here, `site.config["baseurl"]` _can_ be `nil`, but on master, [`parts.compact`](https://github.com/jekyll/jekyll/blob/b0136fccfab51fd7885207fa79bc3dbf3f64c6f4/lib/jekyll/filters/url_filters.rb#L26-L29) is always a no-op. @benbalter, does that invalidate the PR on master?